### ISSUE TITLE
Improve "control" backend functionality

### ISSUE THREAD

**Issue by raphaelm** [venueless/venueless#225]
- [x] option to note down capacity planning info (when an event is taking place and with how many people, noting that event days do not need to be consecutive)
- [x] management of JanusServer/BBBServer/TurnServer instances
- [ ] tests
- [x] 2FA
- [x] maybe a list and detail view of the Feedback model to inspect janus logs

**Comment by raphaelm** [comment]
2fa is now implemented

**Comment by saschafoerster** [comment]
Is it already in stable? I tried to activate it, but didn't succeed.

**Comment by rixx** [comment]
Multifactor is in stable, yes – the configuration page should be linked on your profile page in /control, as "Multi-factor authentication settings"

**Comment by saschafoerster** [comment]
I really don't see it. I updated Venueless to the latest stable version just right now, restarted and I don't find any option in control to activate multi factor, neither in the user profile or in any other menu.

**Comment by rixx** [comment]
I'm really not sure what to say, the option has been htere for a month now for me. You can see it in the attached screenshot:
![2022-01-19T20:43:26+01:00](https://user-images.githubusercontent.com/2657547/150202611-a55961be-b715-4ab5-a173-1a72519aa54a.png)

**Comment by saschafoerster** [comment]
I found the issue. When I switched in the docker-compose.yml from to dev, the multi-factor-authentication was there. 
```
#    image: "venueless/venueless:latest"
image: "venueless/venueless:dev"
```
Then I looked into my update workflow: 

```
sudo docker pull venueless/venueless:stable 
# sudo docker pull venueless/venueless:dev 
sudo docker-compose down
sudo docker-compose up -d
sudo docker exec -it venueless_venueless_1 venueless migrate
```
Probably I didn't get updates for a while because I asked docker to pull venueless stable, but then :latest was used in docker-compose up -d when using the (oldish) docker-compose.yml, where :latest was referenced. 

Is there someplace to see version-numbers in Venueless backend? 😳

**Comment by rixx** [comment]
The trouble is that venueless doesn't even have any version number it could show :/

**Comment by saschafoerster** [comment]
Oh and I can't use the 2-FA-method offered. I am using the "six-numbers-method"with a password-manager, where I either scan a QR-code or copy a generation link, to create the numbers. I only see FIDO or other methods with USB-sticks. Or am I doing something wrong? :)

**Comment by raphaelm** [comment]
Yes, we only support WebAuthn-style 2FA right now, which usually means FIDO2 security keys, but some browsers on some platforms also can do it natively (i.e. treating the internal security enclave of your notebook as a security key).

The "six-numbers-method" (formally TOTP) is way less secure for two main reasons:

1. As you already mentioned some people keep the TOTP secret in the same place as the password (in the password manager), so if your password manager is compromised, 2FA won't help. With a hardware security key, it will.
2. If someone sends you a phishing email asking you click on https://admin.venueless.org.evilsite.com and you fall for it you might enter your password, and you might also enter your six-digit-token and the attacker can just relay it to the original site. With WebAuthn, the process stops there, since the cryptographic scheme hashes in the domain name and it will just fail if you're on the wrong domain.

So basically, TOTP still protects you against *a service* leaking your password, or against some *guessing* your password, but not against compromise of your system and not against phishing. At least phishing is a very real attack vector.

For this reason, our internal guidelines for systems *we* operate do not allow TOTP.

The Django library we used to implement this generally supports TOTP, but we stripped it out. If someone needs it, it could be put back in but hidden behind a server-level configuration flag. Not sure if we'll invest time here, but I'd certainly accept a pull request for that.
